library/java/src/io/envoyproxy/envoymobile/engine/AndroidEngine.java

```diff
@@ -2,22 +2,21 @@
 
 import android.content.Context;
 import android.net.ConnectivityManager;
-import io.envoyproxy.envoymobile.engine.types.EnvoyStatus;
 
 public class AndroidEngine {
 
   // Internal reference to helper object used to load and initialize the native library.
   // Volatile to ensure double-checked locking works correctly.
   private static volatile AndroidEngine loader = null;
 
-  private final Engine engine;
+  private final EnvoyEngine envoyEngine;
 
   // Private helper class used by the load method to ensure the native library and its
   // dependencies are loaded and initialized at most once.
   private AndroidEngine(Context context) {
     System.loadLibrary("envoy_jni");
-    initialize((ConnectivityManager)context.getSystemService(Context.CONNECTIVITY_SERVICE));
-    engine = new EnvoyEngine();
+    initialize((ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE));
+    envoyEngine = new EnvoyEngineImpl();
```

Comment:
Assuming an upcoming PR will allow for specifying a custom/mockable implementation of `EnvoyEngine`

Reply:
Yup. This isn't as important at the moment because we're not "using" the new interfaces within our example applications. I plan to take some time to update this when we do a full end to end wire